runtime/compiler/control/J9Options.cpp

```diff
@@ -2474,6 +2768,322 @@ J9::Options::printPID()
+   if (options->_induceOSR)
+      options->_induceOSR = (char *)((uint8_t *)&(options->_induceOSR) + (ptrdiff_t)options->_induceOSR);
+
+   // receive rtResolve
+   // NOTE: this relies on rtResolve being the last option in clientOptions
+   // the J9JIT_RUNTIME_RESOLVE flag from JITServer client
+   // On JITServer, we store this value for each client in ClientSessionData
+   bool rtResolve = (bool) *((uint8_t *) options + clientOptionsSize - sizeof(bool));
+   // JITSERVER_TODO guards the code that relies on other JITServer unmerged files.
+#if defined(JITSERVER_TODO)
+   compInfoPT->getClientData()->setRtResolve(rtResolve);
+   _reportByteCodeInfoAtCatchBlock = fe->getReportByteCodeInfoAtCatchBlock();
+#endif /* defined(JITSERVER_TODO) */
+   unpackRegex(options->_disabledOptTransformations);
+   unpackRegex(options->_disabledInlineSites);
+   unpackRegex(options->_disabledOpts);
+   unpackRegex(options->_optsToTrace);
+   unpackRegex(options->_dontInline);
+   unpackRegex(options->_onlyInline);
+   unpackRegex(options->_tryToInline);
+   unpackRegex(options->_slipTrap);
+   unpackRegex(options->_lockReserveClass);
+   unpackRegex(options->_breakOnOpts);
+   unpackRegex(options->_breakOnCreate);
+   unpackRegex(options->_debugOnCreate);
+   unpackRegex(options->_breakOnThrow);
+   unpackRegex(options->_breakOnPrint);
+   unpackRegex(options->_enabledStaticCounterNames);
+   unpackRegex(options->_enabledDynamicCounterNames);
+   unpackRegex(options->_counterHistogramNames);
+   unpackRegex(options->_verboseOptTransformationsRegex);
+   unpackRegex(options->_packedTest);
+   unpackRegex(options->_memUsage);
+   unpackRegex(options->_classesWithFolableFinalFields);
+   unpackRegex(options->_disabledIdiomPatterns);
+
+   return options;
+   }
+
+std::string
+J9::Options::packLogFile(TR::FILE *fp)
+   {
+   if (fp == NULL)
+      return "";
+   const size_t size = 4096; // 4kb
+   char buf[size + 1];
+   ::rewind(fp->_stream);
+   std::string logFileStr("");
+   int readSize;
+   do {
+      readSize = ::fread(buf, 1, size, fp->_stream);
+      buf[readSize] = '\0';
+      logFileStr.append(buf);
+   } while (readSize == size);
+
+   logFileStr.append("</jitlog>\n");
+   return logFileStr;
+   }
+
+uint8_t *
+J9::Options::appendContent(char * &charPtr, uint8_t * curPos, size_t length)
+   {
+   if (charPtr == NULL)
+      return curPos;
+   // copies charPtr's content to the location pointed by curPos
+   memcpy(curPos, charPtr, length);
+   // then compute the offset from address of charPtr to curPos and store it to charPtr
+   charPtr = (char *)(curPos - (uint8_t *)&(charPtr));
+   // update current position
+   return curPos += length;
+   }
+
+TR_Debug *createDebugObject(TR::Compilation *);
+// JITServer: create a log file for each client compilation request
+// Used by JITServer
+// Side effect: set _logFile
```

Comment:
I would like more comments on the high level functionality of `setLogFileForClientOptions` `closeLogFileForClientOptions` and `writeLogFileFromServer`